src/models/payments.js

```diff
@@ -1,60 +1,18 @@
 'use strict'
 
 const _ = require('lodash')
-const BigNumber = require('bignumber.js')
 const testPaymentExpiry = require('../lib/testPaymentExpiry')
 const log = require('../common').log('payments')
 const executeSourceTransfers = require('../lib/executeSourceTransfers')
-const UnacceptableConditionsError =
-  require('../errors/unacceptable-conditions-error')
-const UnacceptableRateError = require('../errors/unacceptable-rate-error')
-const NoRelatedSourceCreditError =
-  require('../errors/no-related-source-credit-error')
-const NoRelatedDestinationDebitError =
-  require('../errors/no-related-destination-debit-error')
 const UnrelatedNotificationError =
   require('../errors/unrelated-notification-error')
 const AssetsNotTradedError = require('../errors/assets-not-traded-error')
-const backend = require('../services/backend')
 const routeBuilder = require('../services/route-builder')
 
-function sourceConditionSameAsAllDestinationConditions (
-  sourceTransfer, destinationTransfers) {
-  return _.every(destinationTransfers,
-    function (destinationTransfer) {
-      return _.isEqual(sourceTransfer.execution_condition,
-        destinationTransfer.execution_condition)
-    })
-}
-
-function validateExecutionConditions (payment) {
-  log.debug('validating execution conditions')
-  // We need to have confidence that the source transfers will actually happen.
-  // So each one has to match the condition of all of the destination transfers.
-  // (So we can just copy the condition's fulfillment.)
-
-  // Note that implementing this correctly is VERY IMPORTANT for the connector
-  // to make sure they get paid back and avoid getting screwed
-
-  // If this logic changes, make sure to change the logic in
-  // validateExpiry as well
-
-  const valid = _.every(payment.source_transfers, function (sourceTransfer) {
-    return sourceConditionSameAsAllDestinationConditions(
-      sourceTransfer, payment.destination_transfers)
-  })
-
-  if (!valid) {
-    throw new UnacceptableConditionsError("Each of the source transfers' " +
-      'execution conditions must match all of the destination ' +
-      "transfers' conditions")
-  }
-}
-
-function * validateExpiry (payment, config) {
+function * validateExpiry (sourceTransfer, destinationTransfer, config) {
   // TODO tie the maxHoldTime to the fx rate
   // TODO bring all these loops into one to speed this up
-  const tester = yield testPaymentExpiry(config, payment)
+  const tester = yield testPaymentExpiry(config, sourceTransfer, destinationTransfer)
   tester.validateNotExpired()
   if (tester.isAtomic()) {
     tester.validateMaxHoldTime()
```

Comment:
`tester.validateMaxHoldTime()` is called in both branches of the `if...else` so it should just be taken out and moved after  `tester.validateNotExpired()`